### PR TITLE
Issue 27:   Added DomianRegion Model

### LINKED ISSUE #27: Define Region based domain strategy
Determine the best way to set domains via a region defined at the time of merchant file ingest 

- [x] Add Region model
- [x] Associate Region model with merchant files
- [x] Associate Region model with merchants
- [x] Associate Region model with domains

### PR DESCRIPTION
**Pull request by andrewnisbet74**
Added a new model to represent the relationship between the domain and region models this will be used to build the domain list in the event that a country code isn't provided as part of the merchant data.  There are a number of associations and spec updates here to support this change. 

closes #27

### CODE REVIEW

**Comment by andrewnisbet74** [comment]
@DeathMetalCoder

File: app/jobs/url_crawler_job.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  include Sidekiq::Worker
+
+  def perform(url_id)
+  	url = Url.find(url_id)
+  	content = url.query_common_crawl
+    unless content.nil?
+       #move on to the next step 
+    end
+  end
+end
```
Comment (DeathMetalCoder) [inline app/jobs/url_crawler_job.rb]: There is a directory in (app) that is called "workers" that this should go in.

**Comment by DeathMetalCoder** [comment]
@andrewnisbet74  A lot of the models in this PR are going to conflict with my PR when merged. I would also like to talk about the relationship between region , Merchat, MerchatFile, and DomainRegion. I don't think I am onboard with the way it is setup in this PR. We should have a quick chat about this and look at my current PR.

**Comment by andrewnisbet74** [comment]
@DeathMetalCoder  ok, what are you concerns/issues here?

**Pull request merged**